### PR TITLE
Update RIC Fusion for known Gathers

### PR DESCRIPTION
**Pull request by GlebKazantaev**
### Description
RIC fusion was updated to fuse Gathers (RIC) for cases when order is unknown. This cases appears when we apply RIC preprocessing for dynamic inputs, so this Gather (RIC) operation are marked with special flag which indicates that this operation is RIC operation and can be fused without any bad consequences.

### CODE REVIEW

File: src/core/src/preprocess/preprocess_steps_impl.cpp [hunk 1]
```diff
@@ -362,7 +364,11 @@ void PreStepsList::add_reverse_channels() {
     m_actions.emplace_back([](const std::vector<Output<Node>>& nodes,
                               const std::shared_ptr<Model>& function,
                               PreprocessingContext& context) {
-        return reverse_channels(nodes, function, context);
+        auto resp = reverse_channels(nodes, function, context);
+        auto outputs = std::get<0>(resp);
+        OPENVINO_ASSERT(outputs.size() == 1, "Internal error: reverse_channels returned unexpected number of outputs");
+        set_is_reverse_input_channels(outputs.at(0).get_node_shared_ptr());
```
Comment (jane-intel) [inline src/core/src/preprocess/preprocess_steps_impl.cpp:370]: is it ok? we pass nodes here. maybe there are several RICs there?

Comment (GlebKazantaev) [reply src/core/src/preprocess/preprocess_steps_impl.cpp:369]: Yes, it is ok.  https://github.com/openvinotoolkit/openvino/blob/231dcb75f5c084900a375c6c71c3cb056db0e39e/src/core/src/preprocess/preprocess_steps_impl.cpp#L378

File: src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp [hunk 2]
```diff
@@ -372,17 +386,24 @@ class Convolution : public ngraph::pass::MatcherPass {
         auto input_p = pattern::any_input(ric_attr::has<Output<Node>>);
         auto pattern_root = pattern::wrap_type<opset8::Convolution>({input_p,
                                                                      pattern::wrap_type<opset8::Constant,
-                                                                                        opset8::FakeQuantize>()});
+                                                                                        opset8::FakeQuantize>(pattern::has_static_dim(1/*output channel*/))});
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp:389]: ?

**Review by jane-intel — APPROVED** [review]
Please don't forget to revert main.py change

**Comment by GlebKazantaev** [comment]
All MO Arg tests are no working with enabled by default ONNX FE.
image

_TestsReverseInputChannels::test_ric_preproc_preproc - fail is expected_

@mitruska @jane-intel  FYI ^^^

**Pull request merged**